storm-core/src/clj/backtype/storm/daemon/executor.clj

```diff
@@ -445,6 +445,32 @@
     ret
     ))
 
+(defn init-batching-buffer [worker-context component-id]
+  (let [batchSize (.get_batch_size (.getComponentCommon worker-context component-id))]
+    (if (> batchSize 1)
+      (let [consumer-ids (flatten (for [cids (vals (.getTargets worker-context component-id))] (keys cids)))
+            consumer-task-ids (flatten (for [cid consumer-ids :when (not (.startsWith cid "__"))] (into '() (.getComponentTasks worker-context cid))))]
+        (HashMap. (zipmap consumer-task-ids (repeatedly (count consumer-task-ids) #(Batch. batchSize)))))
+      (HashMap.)
+      )))
+
+(defn emit-msg [out-task out-tuple overflow-buffer output-batch-buffer transfer-fn]
+  (let [out-batch (.get output-batch-buffer out-task)]
+    (if out-batch
```

Comment:
Unless I misunderstand, this is going to end up being always true or always false, depending on the result of `.get_batch_size` for the component at startup. In that case, can we decide this when we're setting up the transfer function instead of each time we're emitting a tuple?